x/profiles/types/msgs/msgs.go

```diff
@@ -152,61 +152,153 @@ func (msg MsgRequestDTagTransfer) GetSignBytes() []byte {
-func (msg MsgAcceptDTagTransfer) Type() string { return models.ActionAcceptDtagTransfer }
+func (msg MsgAcceptDTagTransferRequest) Type() string { return models.ActionAcceptDtagTransfer }
 
-func (msg MsgAcceptDTagTransfer) ValidateBasic() error {
+func (msg MsgAcceptDTagTransferRequest) ValidateBasic() error {
 	if len(strings.TrimSpace(msg.NewDTag)) == 0 {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "new dTag can't be empty")
 	}
 
-	if msg.CurrentOwner.Empty() {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid current owner address: %s", msg.CurrentOwner))
+	if msg.Receiver.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid current owner address: %s", msg.Receiver))
 	}
 
-	if msg.ReceivingUser.Empty() {
-		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid receiving user address: %s", msg.ReceivingUser))
+	if msg.Sender.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid receiving user address: %s", msg.Sender))
 	}
 
-	if msg.ReceivingUser.Equals(msg.CurrentOwner) {
+	if msg.Sender.Equals(msg.Receiver) {
 		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "the receiving user and current owner must be different")
 	}
 
 	return nil
 }
 
 // GetSignBytes encodes the message for signing
-func (msg MsgAcceptDTagTransfer) GetSignBytes() []byte {
+func (msg MsgAcceptDTagTransferRequest) GetSignBytes() []byte {
+	return sdk.MustSortJSON(MsgsCodec.MustMarshalJSON(msg))
+}
+
+// GetSigners defines whose signature is required
+func (msg MsgAcceptDTagTransferRequest) GetSigners() []sdk.AccAddress {
+	return []sdk.AccAddress{msg.Receiver}
+}
+
+// MsgRefuseDTagTransferRequest represent a DTag request rejection
+type MsgRefuseDTagTransferRequest struct {
+	Receiver sdk.AccAddress `json:"receiver" yaml:"receiver"`
+	Sender   sdk.AccAddress `json:"sender" yaml:"sender"`
+}
+
+// NewMsgRefuseDTagTransferRequest is a constructor for MsgRefuseDTagTransferRequest
+func NewMsgRefuseDTagTransferRequest(receiver, sender sdk.AccAddress) MsgRefuseDTagTransferRequest {
+	return MsgRefuseDTagTransferRequest{
+		Receiver: receiver,
+		Sender:   sender,
+	}
+}
+
+// Route should return the name of the module
+func (msg MsgRefuseDTagTransferRequest) Route() string { return models.RouterKey }
+
+// Type should return the action
+func (msg MsgRefuseDTagTransferRequest) Type() string { return models.RefuseDTagTransferRequest }
+
+func (msg MsgRefuseDTagTransferRequest) ValidateBasic() error {
+	if msg.Sender.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid sender address: %s", msg.Sender))
+	}
+
+	if msg.Receiver.Empty() {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidAddress, fmt.Sprintf("invalid receiver address: %s", msg.Receiver))
+	}
+
+	if msg.Sender.Equals(msg.Receiver) {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "the sender and receiver addresses must be different")
```

Comment:
Let's change this to make it coherent with the other errors

```suggestion
		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "the sender and receiver must be different")
```